pywbem_mock/_mainprovider.py

```diff
@@ -60,18 +64,21 @@
 # Maximum Open... timeout if not set by request
 OPEN_MAX_TIMEOUT = 40
 
-# per DSP0200, the default behavior for EnumerateInstance DeepInheritance
+# Per DSP0200, the default behavior for EnumerateInstance DeepInheritance
 # if not set by server.  Default is True.
 DEFAULT_DEEP_INHERITANCE = True
 
+# None of the request method names conform since they are camel case
+# pylint: disable=invalid-name
+
 
-class CIMRepository(ResolverMixin, object):
+class MainProvider(ResolverMixin, object):
     """
```

Comment:
The first sentence of the docstring is quite complicated and is still written around the term CIM repository.
In the second sentence, pywbem class names are used in plural, i.e. "CIMClasses".
Also, a definition of what a "provider" is in pywbem would be useful to have somewhere.

To address all of that, I suggest to replace the first two paragraphs with:

```
A CIM provider in the mock support of pywbem handles operations defined in DSP0200
and implements them by working against a CIM repository. Because the CIM repository
has a uniform interface, most operations can be implemented in a generic way, and
only those operations that create, delete or modify CIM instances have a need to
have user-defined providers.

The MainProvider class is the main CIM provider in the mock support of pywbem
and handles all those operations that do not have a need to be implementable by
users. Specifically, it implements:

* All operations on CIM classes.
* All operations on CIM qualifier declarations.
* All read operations on CIM instances (i.e. get, traditional and pull enumerations,
  association operations).

Operations that create, delete or modify CIM instances have a default implementation
in the DefaultInstanceProvider class and can be overwritten by users on a per class basis.

For more details, see mocksupport.rst.
```

Reply:
Replaced with a few minor changes.  DONE